### PR TITLE
Update relayer with latest tangle-runtime

### LINKED ISSUE #597: [TASK] Update Relayer with Latest Tangle Runtime
**Step zero**:
- [x] Update Relayer dependencies and crates
- [x] Disable all old tests that are releated to the Bridge (for now)

**Step one**:
- [x] Update webb-rs with the new tangle runtime (remove anything releated to DKG, and unused protocol-substrate)
- [x] Add Tangle testnet and mainnet runtime.

**Step two**:
- [x] Update Relayer Listeners to work over Jobs Event instead of DKGs
- [x] Listen for Job Results and events
- [x] Make sure the handlers are up-to-date.

**Step three**
- [ ] Add a dummy handler as a test that verifies the data and signature from TSS Phase two are signed by the Key from phase one.

### LINKED ISSUE #601: [TASK] Refactor proposal signing backend for relayer
### Summary
Refactor  proposal signing backend from relayer configuration.

- [x] Refactor Proposal signing backend 
- [x] update configs and tests

Add Proposal signing backend to the main level and add Signingrules backend config.

```rust
/// Enumerates the supported different signing backends configurations.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "type")]
pub enum ProposalSigningBackendConfig {
    /// Uses signing rules contract to vote and submit proposals for signing.
    Dkg(DkgProposalSigningBackendConfig),
    /// Uses the Private Key of the current Governor to sign proposals.
    Mocked(MockedProposalSigningBackendConfig),
}

/// DkgProposalSigningBackendConfig represents the configuration for the DKG signing backend.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all(serialize = "camelCase", deserialize = "kebab-case"))]
pub struct DkgProposalSigningBackendConfig {
    /// The address of this contract on this chain.
    pub address: Address,
    /// Phase1 Job Id
    pub phase1_job_id: [u8; 32],
}

/// MockedSigningBackendConfig represents the configuration for the Mocked signing backend.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all(serialize = "camelCase", deserialize = "kebab-case"))]
pub struct MockedProposalSigningBackendConfig {
    /// The private key of the current Governor.
    #[serde(skip_serializing)]
    pub private_key: PrivateKey,
}


```

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- Update the relayer with the latest tangle runtime.
- Update event watchers.
- Submit AUP for voting.
- Moved ProposaSigningBackend to the main level

## Note
Will make follow-up PRs for todos.

### Reference issue to close (if applicable)
- Closes #597 
- Closes #601 

-----
### Code Checklist 

- [ ] Tested
- [ ] Documented

### CODE REVIEW

File: crates/proposal-signing-backends/src/dkg.rs [hunk 1]
```diff
@@ -35,42 +35,6 @@ impl super::ProposalSigningBackend for DkgProposalSigningBackend {
...
-            .fetch(&resource_id_addrs)
-            .await?;
-        if maybe_resource_id.is_none() {
-            tracing::warn!(
-                resource_id = %hex::encode(resource_id.into_bytes()),
-                "resource id doesn't exist!",
-            );
-            return Ok(false);
-        }
         // all is good!
```
Comment (drewstone) [inline crates/proposal-signing-backends/src/dkg.rs:38]: Let's remove comments like this @salman01zp too

File: crates/proposal-signing-backends/src/dkg.rs [hunk 2]
```diff
@@ -1,77 +1,39 @@
...
+    QueueItem, QueueStore, TransactionQueueItemKey,
+};
 use webb_relayer_store::sled::SledQueueKey;
-use webb_relayer_utils::static_tx_payload::TypeErasedStaticTxPayload;
+use webb_relayer_store::SledStore;
+use webb_relayer_types::EthersClient;
+use webb_relayer_utils::metric;
 
-type DkgClient = OnlineClient<TangleRuntimeConfig>;
 /// A ProposalSigningBackend that uses the DKG System for Signing Proposals.
```
Comment (shekohex) [inline crates/proposal-signing-backends/src/dkg.rs:15]: Update comments
```suggestion
/// A ProposalSigningBackend that uses the Signing Rules Contract for Signing Proposals.
```

File: services/webb-relayer/src/service/tangle.rs [hunk 3]
```diff
@@ -83,111 +68,46 @@ async fn ignite_tangle_runtime(
 /// * `client` - DKG client
 /// * `chain_id` - An u32 representing the chain id of the chain
 /// * `store` -[Sled](https://sled.rs)-based database store
-pub fn start_dkg_proposal_handler(
+pub fn start_job_result_watcher(
```
Comment (shekohex) [inline services/webb-relayer/src/service/tangle.rs:71]: Comments.

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good, just update the code comments to make it match what the code is actually doing.

**Review by shekohex — APPROVED** [review]

**Pull request merged**